<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>842</width>
    <height>870</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>842</width>
    <height>870</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Image Filter App</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="4" column="0" colspan="3">
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="1" column="1">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QComboBox" name="filterEditorSelection">
          <item>
           <property name="text">
            <string>Inversion</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Brightness Correction</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Contrast Enhancement</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addPoint">
          <property name="text">
           <string>Add Point</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="removePoint">
          <property name="text">
           <string>Remove Point</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="resetPoints">
          <property name="text">
           <string>Reset Points</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveCustomFilter">
          <property name="text">
           <string>Save Filter</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="applyCustomFilter">
          <property name="text">
           <string>Apply to Image</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="FilterEditorWidget" name="filterEditorWidget" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="0" colspan="3">
     <layout class="QGridLayout" name="imagesGrid">
      <item row="1" column="1">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Filtered Image</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="applyButton">
        <property name="text">
         <string>Apply Filter</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Original Image</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="filteredImageLabel">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QComboBox" name="filterComboBox">
        <property name="editable">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Inversion</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Brightness Correction</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Contrast Enhancement</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Gamma Correction</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Blur</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Gaussian Blur</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sharpen</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Edge Detection</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Emboss</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="originalImageLabel">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="loadButton">
      <property name="text">
       <string>Load Image</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="resetButton">
      <property name="text">
       <string>Reset Image</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="3">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="3" column="1">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="1">
         <widget class="QSpinBox" name="ditheringRedColorsSpinBox">
          <property name="minimum">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="ditheringGreenColorsSpinBox">
          <property name="minimum">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QSpinBox" name="ditheringBlueColorsSpinBox">
          <property name="minimum">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Colors per RGB channel:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="applyDithering">
          <property name="text">
           <string>Apply Dithering</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Maximum number of colors:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="quantizationMaximumColors">
          <property name="minimum">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="applyQuantization">
          <property name="text">
           <string>Apply Quantization</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="saveButton">
      <property name="text">
       <string>Save Image</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>842</width>
     <height>38</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FilterEditorWidget</class>
   <extends>QWidget</extends>
   <header>gui/filtereditorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
